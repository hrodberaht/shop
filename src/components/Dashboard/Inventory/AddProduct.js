import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { reset } from 'redux-form';
import ConnectedAddProductForm from './AddProductForm';
import { addProduct } from '../../../store/products/actionCreators';
import { getAuthToken } from '../../../store/auth/selectors';

export class AddProduct extends Component {
  submit = (values) => {
    const { addProductToDB, token, clearForm } = this.props;
    addProductToDB(values, token);
    clearForm('addProduct');
  };

  render() {
    return (
      <div className="add-product-inventory">
        <ConnectedAddProductForm onSubmit={this.submit} />
      </div>
    );
  }
}

const mapStateToProps = state => ({
  token: getAuthToken(state),
});

export default connect(
  mapStateToProps,
  {
    addProductToDB: addProduct,
    clearForm: reset,
  },
)(AddProduct);

AddProduct.propTypes = {
  token: PropTypes.string.isRequired,
  addProductToDB: PropTypes.func.isRequired,
  clearForm: PropTypes.func.isRequired,
};
