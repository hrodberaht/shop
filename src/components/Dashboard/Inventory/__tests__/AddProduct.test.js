import { AddProduct } from '../AddProduct';

const defaultProps = {
  token: '1234',
  addProductToDB: jest.fn(),
};

const setup = buildSetup(AddProduct, defaultProps);

describe('<AddProduct />', () => {
  it('should render without crash', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });
  it('should call addProductToDb when submit', () => {
    const { wrapper } = setup();
    wrapper.instance().submit();
    expect(wrapper.instance().props.addProductToDB).toHaveBeenCalled();
  });
});
