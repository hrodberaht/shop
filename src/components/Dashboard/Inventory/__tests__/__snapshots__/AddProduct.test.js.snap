// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddProduct /> should render without crash 1`] = `
<div>
  <Connect(ReduxForm)
    onSubmit={[Function]}
  />
</div>
`;
